src/pipelines/ecmwf.py

```diff
@@ -0,0 +1,197 @@
+
+load_dotenv()
+
+import ocha_stratus as stratus  # noqa
+
+
+logger = logging.getLogger(__name__)
+
+
+def retrieve_ecmwf(
+    start_date, end_date, stage="local", save_to_blob=False, save_dir=None, use_cache=True
+):
+    """
+    Download ECMWF data, upload raw to azure if needed and return loaded Dataset
+    """
+
+    logger.info(f"Retrieving from ECMWF {start_date} to {end_date} ...")
+    filename = f"ECMWF_{start_date.date()}_{end_date.date()}.csv"
+    file_path = f"{save_dir}/" + filename
+
+    if os.path.exists(file_path):
+        logger.info(f"Using file downloaded in {file_path}...")
+        df = pd.read_csv(file_path, parse_dates=["issued_time", "valid_time"])
+    else:
+        if stage == "local":
+            save_dir = Path(save_dir) if save_dir else Path("temp")
+            os.makedirs(save_dir, exist_ok=True)
+
+        date_list = rrule.rrule(
+            rrule.HOURLY,
+            dtstart=start_date,
+            until=end_date + timedelta(hours=12),
+            interval=12,
+        )
+
+        dfs = []
+        for date in date_list:
+            logger.info(f"Processing for {date}...")
+            raw_file = download_hindcasts(
+                date, "storm", use_cache, False, stage
+            )
+            if raw_file:
+                df = _process_cxml_to_df(raw_file, stage, "storm")
+                if df is not None:
+                    dfs.append(df)
+        if len(dfs) > 0:
+            df = pd.concat(dfs)
+            df.to_csv(file_path, index=False, na_rep=None)
+
+            logger.info(f"Successfully wrote ECMWF data to {file_path}.")
+            return df
+        logger.error("No data available for input dates")
+        return None
+        """
+        df = lens.ecmwf_storm.load_hindcasts(
+            start_date=start_date,
+            end_date=end_date,
+            stage=stage,
+            temp_dir=save_dir,
+            use_cache=False
+        )
+        """
+
+
+    if save_to_blob:
+        logger.info(f"Uploading {file_path} to Azure blob in {stage}...")
+        with open(file_path, "rb") as file:
+            data_to_upload = file.read()
+        stratus.upload_blob_data(
+            container_name="storm",
+            data=data_to_upload,
+            blob_name=f"ecmwf/{filename}",
+            stage=stage,
+        )
+        logger.info("Successfully uploaded to blob.")
+
+    return df
+
+
+def process_tracks(dataset, engine, chunksize):
+    """
+    Retrieve 'best' and 'provisional' tracks and upload them to the database
+    """
+    logger.info("Extracting tracks...")
+    tracks_geo = lens.ecmwf_storm.get_tracks(dataset)
+
+    # In order to comply with the type of object we can apply this function to each geometry
+    # and then run the upsert or use to_postgis to a temporary table instead of to_sql and
+    # then run another query to do the upsert
+    logger.info("Transforming geometry...")
+    tracks_geo["geometry"] = tracks_geo["geometry"].to_wkt()
```

Comment:
Is it possible to address this warning that comes up? 

`2025-10-29 13:51:53 - src.pipelines.ecmwf - INFO - Transforming geometry...
/Workspace/Repos/.internal/2dd109d71a_commits/9b4e26bc04d72473b8a4d0d19002a2a4b4c98dd7/src/pipelines/ecmwf.py:107: UserWarning: Geometry column does not contain geometry.
  tracks_geo["geometry"] = tracks_geo["geometry"].to_wkt()`